src/Entity/NumericEntityIdProvider.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Wikibase\DataModel\Entity;
+
+/**
+ * Common interface for EntityIds that can be converted to numbers, and back from the entity type
+ * and the number. Entity types that do not met this criteria (e.g. when page titles or file names
+ * are used as IDs) should not implement this interface. Never return a fallback value like 0
+ * that's the same for multiple IDs!
+ *
+ * Entity types are not required and not guaranteed to implement this interface. Use the full string
+ * serialization whenever you can and avoid using numeric IDs.
+ *
+ * @since 6.1
+ *
+ * @see EntityId::getSerialization
+ */
+interface NumericEntityIdProvider {
+
+	/**
+	 * @since 6.1
+	 *
+	 * @return int|float Numeric representation of the EntityId as a positive integer number in the
+	 *  [1..PHP_INT_MAX] range. A float when exceeding this range (e.g. when exceeding 2,147,483,647
+	 *  on a 32 bit system).
```

Comment:
>_> such PHP

Will using `int` s type hint here cause problems on 32 bit systems, even if PHP is using a float for it? ie, will `is_int` checks fail in this special case? (Or the real `int` type hint in modern PHP for that matter)